File: surf-cloud-api/src/main/java/dev/slne/surf/cloud/api/netty/exception/SurfNettyListenerException.java
package dev.slne.surf.cloud.api.netty.exception;

import java.io.Serial;
import lombok.experimental.StandardException;

@StandardException
public abstract class SurfNettyListenerException extends SurfNettyException{

  @Serial
  private static final long serialVersionUID = 7748641971923464222L;
}

File: surf-cloud-api/surf-cloud-api-common/src/main/kotlin/dev/slne/surf/cloud/api/common/netty/exception/SurfNettyListenerException.kt
package dev.slne.surf.cloud.api.common.netty.exception

import java.io.Serial

/**
 * Base exception for listener-related Netty errors.
 */
abstract class SurfNettyListenerException : SurfNettyException {
    constructor() : super()
    constructor(cause: Throwable?) : super(cause)
    constructor(message: String?) : super(message)
    constructor(message: String?, cause: Throwable?) : super(message, cause)


    companion object {
        @JvmStatic
        @Serial
        private val serialVersionUID = 7748641971923464222L
    }
}
